pkg/dockerregistry/server/pullthroughblobstore.go

```diff
@@ -123,130 +54,57 @@ func (r *pullthroughBlobStore) proxyStat(ctx context.Context, retriever importer
 // success response with no actual body content.
 // [1] https://docs.docker.com/registry/spec/api/#existing-layers
 func (pbs *pullthroughBlobStore) ServeBlob(ctx context.Context, w http.ResponseWriter, req *http.Request, dgst digest.Digest) error {
-	store, ok := pbs.digestToStore[dgst.String()]
-	if !ok {
+	if _, err := pbs.BlobStore.Stat(ctx, dgst); err == nil {
```

Comment:
There's no need for the `Stat`. It's the first thing the `pbs.BlobStore.ServeBlob` does. Just handle the error case.